src/vulkan-renderer/wrappers/instance.cpp

```diff
@@ -0,0 +1,174 @@
+        spdlog::warn("Without them, most bugs are literally impossible to find.");
+        spdlog::warn("Even worse, your software might work on your machine but does crash on other systems!");
+    }
+#endif
+
+    // Add requested instance layers to wishlist.
+    if (requested_instance_layers.has_value()) {
+        for (const auto &instance_layer : requested_instance_layers.value()) {
+            instance_layers_wishlist.push_back(instance_layer.c_str());
+        }
+    }
+
+    // We have to check which instance layers of our wishlist are available on the current system!
+    // We are not checking for duplicated entries but this is no problem.
+    for (auto current_layer : instance_layers_wishlist) {
+        if (availability_checks->has_instance_layer(current_layer)) {
+            spdlog::debug("Adding '{}' to list of enabled instance layers.", current_layer);
+            enabled_instance_layers.push_back(current_layer);
+        } else {
+#ifdef NDEBUG
+            if (std::string(current_layer).compare(VK_EXT_DEBUG_MARKER_EXTENSION_NAME) == 0) {
+                spdlog::error("You can't use command line argument -renderdoc in release mode.");
+            }
+#else
+            spdlog::error("Requested instance layer '{}' is not available on this system!", current_layer);
+#endif
+        }
+    }
+
+    VkInstanceCreateInfo instance_create_info = {};
+    instance_create_info.sType = VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO;
+    instance_create_info.pNext = nullptr;
+    instance_create_info.flags = 0;
+    instance_create_info.pApplicationInfo = &app_info;
+    instance_create_info.ppEnabledExtensionNames = enabled_instance_extensions.data();
+    instance_create_info.enabledExtensionCount = static_cast<std::uint32_t>(enabled_instance_extensions.size());
+    instance_create_info.ppEnabledLayerNames = enabled_instance_layers.data();
+    instance_create_info.enabledLayerCount = static_cast<std::uint32_t>(enabled_instance_layers.size());
+
+    if (vkCreateInstance(&instance_create_info, nullptr, &instance) != VK_SUCCESS) {
+        throw std::runtime_error("Error: vkCreateInstance failed!");
+    }
+
+    spdlog::debug("Created Vulkan instance successfully.");
+}
+
+Instance::Instance(const std::string &application_name, const std::string &engine_name, const std::uint32_t application_version,
+                   const std::uint32_t engine_version, const std::uint32_t vulkan_api_version)
+    : Instance(application_name, engine_name, application_version, engine_version, vulkan_api_version, std::nullopt, std::nullopt, true, false) {
+    spdlog::debug("No instance extensions or instance layers specified.");
+    spdlog::debug("Validation layers are requested. RenderDoc instance layer is not requested.");
+}
+
+Instance::~Instance() {
+    vkDestroyInstance(this->instance, nullptr);
+    this->instance = VK_NULL_HANDLE;
```

Comment:
I don't think this is needed?

Reply:
I like to reset Vulkan handles to `VK_NULL_HANDLE` because null sticks out more when debugging.